hi-ml-azure/src/health_azure/himl.py

```diff
@@ -685,67 +648,72 @@ def _str_to_path(s: Optional[PathOrString]) -> Optional[Path]:
-    :return: A dictionary in the format "input_name": Input.
+    :param dataset_list: The list of datasets to create data assets for.
+    :raises ValueError: Raised if a data asset has no path.
+    :return: A list of data assets.
     """
-    inputs: Dict[str, Input] = {}
-    for i, input_dataset in enumerate(input_datasets):
-        input_name = f"INPUT_{i}"
-        version = input_dataset.version
-        # if version is None, this function gets the latest version
+    data_assets: List[Data] = []
+    for dataset in dataset_list:
+        version = dataset.version
         logging.info(
-            f"Trying to access data asset {input_dataset.name} version {version}, datastore {input_dataset.datastore}"
+            f"Trying to access data asset {dataset.name} version {version}, datastore {dataset.datastore}"
         )
+
+        # if version is None, this function gets the latest version
         data_asset: Data = _get_or_create_v2_data_asset(
             ml_client,
-            input_dataset.datastore,
-            input_dataset.name,
+            dataset.datastore,
+            dataset.name,
             version=str(version) if version else None,
         )
         if not data_asset.path:
             raise ValueError(f"Data asset {data_asset.id} has no path.")
-        # Some mismatches with the documentation here:
-        # data_path = data_asset.id: This works in some jobs, but in other gives a
-        # DataAccessError(InvalidInput { message: "invalid uri format", source: None }))
-        # Unclear what the difference is between failing and successful jobs
-        # Alternative: data_path = f"azureml:{data_asset.name}:{version}"
-        # This does not work at all, neither with v1 nor v2 data assets. In both cases, we get
-        # InvalidInput { message: "invalid uri format"
-        data_path = data_asset.path
-
-        inputs[input_name] = Input(
-            # Data assets can be of type "uri_folder", "uri_file", "mltable", all of which are value types in Input
-            type=data_asset.type,  # type: ignore
-            path=data_path,
-            mode=InputOutputModes.MOUNT if input_dataset.use_mounting else InputOutputModes.DOWNLOAD
-        )
-    return inputs
+
+        data_assets.append(data_asset)
+
+    return data_assets
+
+
+def create_v2_inputs(ml_client: MLClient, input_datasets: List[DatasetConfig]) -> Dict[str, Input]:
+    """
+    Create a dictionary of Azure ML v2 Input objects, required for passing input data in to an AML job
+
+    :param ml_client: An MLClient object.
+    :param input_datasets: A list of DatasetConfigs to convert to Inputs.
+    :return: A dictionary in the format "input_name": Input.
+    """
+    input_assets = get_data_assets_from_configs(ml_client, input_datasets)
+    # Data assets can be of type "uri_folder", "uri_file", "mltable", all of which are value types in Input
+    return {
+        f"INPUT_{i}": Input(  # type: ignore
```

Comment:
I'm a big opponent of repeated literal strings. The "INPUT" part here must match the regex part in line 1088, so it would be better to define a constant. Or even simpler, name your inputs "0", "1", etc.